extensions/panache/mongodb-panache-common/deployment/src/test/java/io/quarkus/mongodb/panache/common/MongoDatabaseResolverTest.java

```diff
@@ -0,0 +1,199 @@
+package io.quarkus.mongodb.panache.common;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.bson.Document;
+import org.bson.codecs.pojo.annotations.BsonId;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+
+import io.quarkus.mongodb.panache.common.reactive.runtime.ReactiveMongoOperations;
+import io.quarkus.mongodb.panache.common.runtime.MongoOperations;
+import io.quarkus.mongodb.reactive.ReactiveMongoClient;
+import io.quarkus.mongodb.reactive.ReactiveMongoCollection;
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.mongodb.MongoReplicaSetTestResource;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@QuarkusTestResource(MongoReplicaSetTestResource.class)
+@DisabledOnOs(OS.WINDOWS)
+public class MongoDatabaseResolverTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addAsResource(new StringAsset(
+                            "quarkus.arc.remove-unused-beans=false\n" +
```

Comment:
Can you remove the `quarkus.arc.remove-unused-beans=false` and check it stills work ?

Reply:
I already had tried before and it doesn't work, because the BuildStep method responsible to make `MongoDatabaseResolver` unremovable is described in `BasePanacheMongoResourceProcessor` that is an abstract class and it implementation is in another project (mongodb-panache)

Reply:
OK, make thanks